id/__init__.py

```diff
@@ -67,6 +68,7 @@ def detect_credential(audience: str) -> Optional[str]:
         detect_github,
         detect_gcp,
         detect_buildkite,
+        detect_env_var,
```

Comment:
Should this be last, or first? I could see this as a handy way to override the token for these other platforms, but not sure if we want to encourage that. cc @woodruffw for thoughts.

Reply:
My initial thought is last (per your point about encouraging it).

From https://github.com/di/id/pull/123#issuecomment-1772819703: maybe `detect_credential` should only contain `detect_gitlab`, and `detect_env_var` could be a separate public API? That way there's no surprising fallback behavior.

Reply:
Thinking about this more, I think we should optimistically assume that most providers won't implement OIDC tokens this way, and that this is something specific to Gitlab. I think `detect_credential` should only contain `detect_gitlab`, which checks for `GITLAB_CI` and uses `_detect_env_var` which becomes a private (I don't think we need a public API for this).

Reply:
Yes agreed with this last comment.  After a quick test it turns out there really won't even be a common `_detect_env_var` at all (because error management is different: we can actually raise on missing variable if we know we are in GitLab context. This significantly improves the misconfiguration case UX)